hstrat/phylogenetic_inference/tree/_impl/_TrieLeafNode.py

```diff
@@ -1,7 +1,12 @@
+import typing
+
 import anytree
 
 from ...._auxiliary_lib import render_to_base64url
 
+if typing.TYPE_CHECKING:
```

Comment:
this might cause a circular import issue if `typing.TYPE_CHECKING` is enabled (that's why TrieInnerNode's not just directly imported to begin with)

Reply:
It shouldn’t - IIRC the type checking thing is there to avoid circular imports but still allow type hinting